client/src/components/Category/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from "react"
+import './style.css'
+import axios from "axios"
+
+class Category extends Component {
+  state = {
+    details: [],
+    userPledges: [],
+    name: "",
+    categories: [],
+
+  }
+  componentDidMount = () => {
+
+    // request to get all categories
+    const userId = 1
+    axios.get(`/api/dashboard`).then(({ data }) => {
+
+      this.setState({
+        details: data
+      })
+    })
+    // request to get pledges that user make it in all categories
+    axios.get(`/api/dasboard_pledges_in_category/${userId}`).then(({ data }) => {
+      this.setState(
+        { userPledges: data })
+    })
+    // request to get all pledges in each category in general 
+    axios.get(`/api/dashboard_pledges_category`)
+      .then(({ data }) => {
+        this.setState({ categories: data })
+      })
+  }
+  // function to push the data to category page
+  moveToCategorActionPage = (id) => {
+    const { history } = this.props
+
+    history.push({
+      pathname: `action-category/${id}`,
+      data: this.state.details
+    })
+  }
+
+  // function to calculate all pledges that user make it in category by filter
+  numberOfPledges = (arr1, item) => {
+    let arr2 = arr1.filter((element) => {
```

Comment:
Expected to return a value at the end of the filter function

Reply:
done